### ISSUE TITLE
Test Suite Issues

### ISSUE THREAD

**Issue by ryansch** [kylemanna/docker-openvpn#194]
Here's a list of trouble I had getting the test suite running locally:
- [ ] The test suite assumes too much about the host system.  I could not run it on the macOS system I use for all of my other docker development.  We should fix this with a build container.  Sample: https://gist.github.com/ryansch/388ce4ffdbdbe906c92dd339aa810ba9
- [ ] The specs currently hardcode the image name to test against to `kylemanna/openvpn`.
- [ ] The `dual-proto` specs require changes to the firewall and assume nothing else is running on `1194` and `443`.  We should look into using a dedicated docker network here.

**Comment by kylemanna** [comment]
Yeah, these all need to be fixed.  The test cases were introduced in a hurry with the philosophy of "rough test cases are better then no test cases" after a number of extreme fringe features started breaking.

The biggest problem with OpenVPN is that for advanced configurations (i.e. IPv6 or dual-proto) it needs to mess with networking options that Docker just doesn't support now.

Also, I'm not familiar with the "build container" concept, can you explain more how it would help?

**Comment by kylemanna** [comment]
>  The specs currently hardcode the image name to test against to kylemanna/openvpn.
This isn't a problem since if you build locally and tag the build as `kylemanna/openvpn` it will use your local copy instead of remotely fetching.
